src/legacy/core_plugins/kibana/server/lib/system_api.js

```diff
@@ -27,5 +28,8 @@ const SYSTEM_API_HEADER_NAME = 'kbn-system-api';
  * @deprecated Use KibanaRequest#isSystemApi
  */
 export function isSystemApiRequest(request) {
-  return !!request.headers[SYSTEM_API_HEADER_NAME];
+  return (
+    !!request.headers[SYSTEM_REQUEST_HEADER_NAME] ||
+    !!request.headers[LEGACY_SYSTEM_API_HEADER_NAME]
+  );
 }
```

Comment:
Note: due to changes in this PR, the Security plugin is no longer using this API, and this is not being used anywhere else. Since it's deprecated we should wait until 8.0 to remove this code.

Reply:
Even though this is exposed via `server.plugins.kibana.systemApi.isSystemApiRequest` we can remove it in a minor version without it being considered a breaking change. We don't currently consider our internal APIs, which third-party plugins might be consuming, as a public interface which shouldn't be removed or changed during minor versions. However, I'll defer to the @elastic/kibana-platform team on whether or not we should keep it around. If we do intend to keep it around, your fixes look 👍

Reply:
>  which third-party plugins might be consuming, as a public interface which shouldn't be removed or changed during minor versions.

sounds correct to me

Reply:
Okay, I'll remove the API then. Thanks folks!